apps/api/src/domain/models/context/repositories/arango-note-query-repository.spec.ts

```diff
@@ -0,0 +1,321 @@
+
+const noteIds = [2, 3, 4].map(buildDummyUuid);
+
+const textForNote = 'the text for a note';
+
+const existingNotes = [1, 2, 3].map((sequenceNumber, index) =>
+    buildTestInstance(EventSourcedNoteViewModel, {
+        id: noteIds[index],
+        note: buildMultilingualTextWithSingleItem(`note #${index}`),
+    })
+);
+
+class WidgetViewModel {
+    readonly type = WIDGET_TYPE;
+
+    id: AggregateId;
+
+    rating: number;
+
+    constructor(dto: DTO<WidgetViewModel>) {
+        if (!dto) return;
+
+        const { id, rating } = dto;
+
+        this.id = id;
+
+        this.rating = rating;
+    }
+}
+
+const generalContext = new GeneralContext();
+
+describe(`ArangoNoteQueryRepository`, () => {
+    let testQueryRepository: INoteQueryRepository;
+
+    let connectionProvider: ArangoConnectionProvider;
+
+    let databaseProvider: ArangoDatabaseProvider;
+
+    let app: INestApplication;
+
+    let widgetDatabase: ArangoDatabaseForCollection<Widget>;
+
+    beforeAll(async () => {
+        const moduleRef = await Test.createTestingModule({
+            imports: [PersistenceModule.forRootAsync()],
+        })
+            .overrideProvider(ConfigService)
+            .useValue(
+                buildMockConfigService(
+                    {
+                        ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                    },
+                    buildConfigFilePath(Environment.test)
+                )
+            )
+            .compile();
+
+        await moduleRef.init();
+
+        app = moduleRef.createNestApplication();
+
+        connectionProvider = app.get(ArangoConnectionProvider);
+
+        connectionProvider.createCollectionIfNotExists(widgetCollectionName);
+
+        databaseProvider = new ArangoDatabaseProvider(connectionProvider);
+
+        testQueryRepository = new ArangoNoteQueryRepository(connectionProvider);
+
+        widgetDatabase = new ArangoDatabaseForCollection<Widget>(
+            new ArangoDatabase(connectionProvider.getConnection()),
+            widgetCollectionName
+        );
+    });
+
+    beforeEach(async () => {
+        await databaseProvider.clearViews();
+    });
+
+    afterAll(async () => {
+        databaseProvider.close();
+    });
+
+    describe(`fetchById`, () => {
+        const existingNote = buildTestInstance(EventSourcedNoteViewModel, {
+            id: noteIds[0],
+            // TODO should this be called text?
+            note: buildMultilingualTextWithSingleItem(textForNote),
+        });
+
+        beforeEach(async () => {
+            await databaseProvider.clearViews();
+
+            await testQueryRepository.create(existingNote);
+        });
+
+        describe(`when the note exists`, () => {
+            it(`should return the note`, async () => {
+                const result = await testQueryRepository.fetchById(existingNote.id);
+
+                expect(result).not.toBe(NotFound);
+            });
+        });
+
+        // TODO when the note does not exist
```

Comment:
Be sure to add this case as well. Simply skip adding the note document in a `beforeEach`, call `fetchById` and expect to get `NotFound`.